Dockerfile

```
@@ -1,45 +1,23 @@
-FROM nvidia/cuda:11.8.0-cudnn8-runtime-ubuntu22.04
+# Stage 1: Build environment
+FROM python:3.12-slim as core
```

NIT based on [this recommendation](https://docs.docker.com/reference/build-checks/from-as-casing/): 
```suggestion
FROM python:3.12-slim AS core
```